projects/plugins/boost/app/modules/cache/Boost_Cache_Settings.php

```diff
@@ -0,0 +1,107 @@
+ * Settings are stored in a file in the boost-cache directory.
+ * This file is shared by autoloaded code and code loaded by advanced-cache.php
+ */
+
+class Boost_Cache_Settings {
+	private $settings   = array();
+	private $last_error = '';
+	private $config_file_path;
+	private $config_file;
+
+	public function __construct() {
+		$this->config_file_path = WP_CONTENT_DIR . '/boost-cache/';
+		$this->config_file      = $this->config_file_path . 'config.php';
+		$this->init_settings();
+	}
+
+	/*
+	 * Load the settings from the config file, if available.
+	 */
+	private function init_settings() {
+
+		if ( ! file_exists( $this->config_file_path ) ) {
+			mkdir( $this->config_file_path, 0755, true ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_mkdir
+		}
+
+		if ( ! file_exists( $this->config_file ) ) {
+			$result = $this->set( array( 'enabled' => false ) );
+			if ( ! $result ) {
+				return false;
+			}
+		}
+
+		$lines = file( $this->config_file );
+		if ( count( $lines ) < 4 ) {
+			$this->last_error = 'Invalid config file';
+			return false;
+		}
+		$settings = json_decode( $lines[3], true );
+		if ( ! is_array( $settings ) ) {
+			$this->last_error = 'Invalid config file';
+			return false;
+		}
+		$this->settings = $settings;
+	}
+
+	/*
+	 * Returns the value of the given setting.
+	 *
+	 * @param string $setting - The setting to get.
+	 * @return mixed - The value of the setting, or false if the setting does not exist. Call get_last_error() to get the error message.
+	 */
+	public function get( $setting ) {
+		if ( ! isset( $this->settings[ $setting ] ) ) {
+			$this->last_error = 'Setting not found';
+			return false;
+		}
+		return $this->settings[ $setting ];
+	}
+
+	/*
+	 * Sets the given settings, and saves them to the config file.
+	 * @param array $settings - The settings to set in a key => value associative
+	 * array. This will be merged with the existing settings.
+	 * Example:
+	 * $result = $this->set( array( 'enabled' => true ) );
+	 *
+	 * @return bool - true if the settings were saved, false otherwise. Call get_last_error() to get the error message.
+	 */
+	public function set( $settings ) {
+		if ( ! is_writable( $this->config_file_path ) ) { // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_is_writable
+			$this->last_error = 'Config file is not writable';
+			return false;
+		}
+
+		$this->settings = array_merge( $this->settings, $settings );
+
+		$contents     = "<?php die();\n/*\n * Configuration data for Jetpack Boost Cache.\n" . json_encode( $this->settings ) . "\n */"; // phpcs:ignore WordPress.WP.AlternativeFunctions.json_encode_json_encode
+		$tmp_filename = $this->config_file . uniqid( uniqid(), true ) . '.tmp';
+		if ( false === file_put_contents( $tmp_filename, $contents ) ) { // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_file_put_contents
+			$this->last_error = 'Could not write to tmp config file';
+			return false;
+		}
```

Comment:
We're now doing this write-to-tmp and rename trick twice. 

I think it's time for this behaviour to go in a utility function.

Reply:
Agreed. The Utils class is in the deactivate PR, so once both of these are merged I'll do that. Issue added in [#35331](https://github.com/Automattic/jetpack/issues/35331)